chaosreliably/controls/experiment.py

```diff
@@ -124,6 +128,85 @@ def before_experiment_control(
         )
 
 
+def before_hypothesis_control(
+    context: Hypothesis,
+    configuration: Configuration = None,
+    secrets: Secrets = None,
+    **kwargs: Any,
+) -> None:
+    """
+    Control run *before* the execution of an Experiments Steady State Hypothesis
+
+    For a given Steady State Hypothesis, the control creates an Experiment Event Entity
+    Context in the Reliably service.
+
+    The Event has the `event_type` of `HYPOTHESIS_START`.
+
+    :param context: Hypothesis object representing the Steady State Hypothesis that is
+        to be executed
+    :param configuration: Configuration object provided by Chaos Toolkit
+    :param secrets: Secret object provided by Chaos Toolkit
+    :param **kwargs: Any additional keyword arguments passed to the control
+    """
+    try:
+        _create_experiment_event(
+            event_type=EventType.HYPOTHESIS_START,
+            name=context["title"],
+            output=None,
+            experiment_run_labels=configuration["chaosreliably"][
+                "experiment_run_labels"
+            ],
+            configuration=configuration,
+            secrets=secrets,
+        )
+    except Exception as ex:
+        logger.debug(
+            f"An error occurred: {ex}, while running the Before Hypothesis control, the"
+            " Experiment execution won't be affected."
+        )
+
+
+def after_hypothesis_control(
+    context: Hypothesis,
+    state: Dict[str, Any],
```

Comment:
Is it worth creating a `State` type in `chaoslib/types.py`

Reply:
Eventually the whole of `chaoslib` will be typed (fully), and when that get's done, we'll have to go through stuff like this and use those types instead

Reply:
Ah okay

Reply:
I'd add the type here, but it's not the library it'd belong in

Plus if we added the type here, then we have a circular dependency where `chaoslib` needs `chaosreliably` for the type, but `chaosreliably` needs `chaoslib`... 🤣

Reply:
Oh dear ... 🤣